syntax = "proto2";

enum etype {
    TYPE1 = 1;
    TYPE2 = 2;
}

message enumtest {
    required int32 a = 1;
    required etype et = 2;
}

message SimpleMessage {
    required int32 lucky_number = 1;
}

message E2_request {
    required uint32 req_id = 1;
}

message E2_dummy_response {
    required uint32 req_id = 1;
    required string mess_string = 2;
    required bool result = 3;
}


message MapFieldEntry {
  required int32 key = 1;
  required int32 value = 2;
}

message map_mess{
    repeated MapFieldEntry map_field = 1;
}

